test/dynamo/test_graph_break_messages.py

```diff
@@ -838,6 +913,105 @@ def fn(x):
+        try:
+            torch.compile(fn, backend="eager", fullgraph=True)()
+        except Exception as exn:
+            e = exn
+
+        self.assertIsNotNone(e)
+
+        msg = "".join(traceback.format_exception(type(e), e, e.__traceback__))
+        # only keep the filenames in the traceback
+        msg = re.sub(r'File ".*\W(\w+\.py)"', 'File "\\1"', msg)
+        # remove line numbers
+        msg = re.sub(r"line (\d+)", "line N", msg)
+        # remove carets
+        msg = re.sub(r"\n\s*~*\^+\n", "\n", msg)
+        self.assertExpectedInline(
+            msg,
+            """\
+Traceback (most recent call last):
+  File "test_graph_break_messages.py", line N, in test_no_internal_compiler_stacktrace
+    torch.compile(fn, backend="eager", fullgraph=True)()
+  File "eval_frame.py", line N, in _fn
+    raise e.with_traceback(None) from None
+torch._dynamo.exc.Unsupported: Call to `torch._dynamo.graph_break()`
+  Explanation: User-inserted graph break. Message: None
+  Hint: Remove the `torch._dynamo.graph_break()` call.
+
+  Developer debug context: Called `torch._dynamo.graph_break()` with args `[]`, kwargs `{}`
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    gn()
+  File "test_graph_break_messages.py", line N, in gn
+    torch._dynamo.graph_break()
+
+Set TORCHDYNAMO_VERBOSE=1 for the internal stack trace. For even more developer context, set TORCH_LOGS="+dynamo"
+
+""",
+        )
+
+    @torch._dynamo.config.patch(verbose=True)
+    def test_internal_compiler_stacktrace_verbose(self):
+        def fn():
+            gn()
+
+        def gn():
+            torch._dynamo.graph_break()
+
+        # assertRaises suppresses the traceback, so manually catch
+        e = None
+        try:
+            torch.compile(fn, backend="eager", fullgraph=True)()
+        except Exception as exn:
+            e = exn
+
+        self.assertIsNotNone(e)
+
+        msg = "".join(traceback.format_exception(type(e), e, e.__traceback__))
+        # only keep the filenames in the traceback
+        msg = re.sub(r'File ".*\W(\w+\.py)"', 'File "\\1"', msg)
+        # remove line numbers
+        msg = re.sub(r"line (\d+)", "line N", msg)
+        msg = re.sub(
+            r"""(?s)Traceback \(most recent call last\):.*
+  File "exc.py", line N, in unimplemented_v2
+    raise Unsupported\(msg\)""",
+            "<Internal traceback>\n",
+            msg,
+        )
+        self.assertExpectedInline(
+            msg,
+            """\
+<Internal traceback>
+
+torch._dynamo.exc.Unsupported: Call to `torch._dynamo.graph_break()`
+  Explanation: User-inserted graph break. Message: None
+  Hint: Remove the `torch._dynamo.graph_break()` call.
+
+  Developer debug context: Called `torch._dynamo.graph_break()` with args `[]`, kwargs `{}`
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    gn()
+  File "test_graph_break_messages.py", line N, in gn
+    torch._dynamo.graph_break()
+
+""",
+        )
+
```

Comment:
can you do an expecttest test for TORCHDYNAMO_VERBOSE=1? You can temporarily set os.env["TORCHDYNAMO_VERBOSE"] ?